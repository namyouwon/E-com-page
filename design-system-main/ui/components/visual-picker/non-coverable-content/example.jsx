// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import classNames from 'classnames';
import _ from '../../../shared/helpers';
import { Fieldset, Legend } from '../../radio-group/base/example';
import { FormElementControl } from '../../form-element';
import {
  VisualPicker,
  VisualPickerContainer
} from '../coverable-content/example';
import { UtilityIcon } from '../../icons/base/example';

export const PackageOne = [
  <span className="slds-text-heading_small" key={_.uniqueId()}>
    Lightning Professional
  </span>,
  <span className="slds-text-title" key={_.uniqueId()}>
    Complete service CRM for teams of any size
  </span>
];

export const PackageTwo = [
  <span className="slds-text-heading_small" key={_.uniqueId()}>
    Lightning Enterprise
  </span>,
  <span className="slds-text-title" key={_.uniqueId()}>
    Everything you need to take support to the next level
  </span>
];

export const PackageThree = [
  <span className="slds-text-heading_small" key={_.uniqueId()}>
    Lightning Unlimited
  </span>,
  <span className="slds-text-title" key={_.uniqueId()}>
    Complete support with enterprise-grade customization
  </span>
];

export let VisualPickerMediaObject = props => (
  <a
    href="#"
    className={classNames(
      'slds-box slds-box_link slds-box_x-small slds-media',
      props.className
    )}
    onClick={e => e.preventDefault()}
  >
    <div className="slds-media__figure slds-media__figure_fixed-width slds-align_absolute-center slds-m-left_xx-small">
      <UtilityIcon className="slds-icon-text-default" symbol="knowledge_base" />
    </div>
    <div className="slds-media__body slds-border_left slds-p-around_small">
      {props.children}
    </div>
  </a>
);

/// ////////////////////////////////////////
// Export
/// ////////////////////////////////////////

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <Fieldset>
        <Legend>Select a plan</Legend>
        <FormElementControl>
          <VisualPickerContainer hasIcon={false}>
            <VisualPicker size="medium" label={PackageOne}>
              <span>
                <span className="slds-text-heading_large">$30</span>
                <span className="slds-text-title">USD/user/month *</span>
              </span>
            </VisualPicker>
            <VisualPicker size="medium" label={PackageTwo}>
              <span>
                <span className="slds-text-heading_large">$150</span>
                <span className="slds-text-title">USD/user/month *</span>
              </span>
            </VisualPicker>
            <VisualPicker size="medium" label={PackageThree}>
              <span>
                <span className="slds-text-heading_large">$300</span>
                <span className="slds-text-title">USD/user/month *</span>
              </span>
            </VisualPicker>
          </VisualPickerContainer>
        </FormElementControl>
      </Fieldset>
    )
  }
];

export let states = [
  {
    id: 'disabled',
    label: 'Disabled option',
    element: (
      <Fieldset>
        <Legend>Select a plan</Legend>
        <FormElementControl>
          <VisualPickerContainer hasIcon={false}>
            <VisualPicker size="medium" label={PackageOne}>
              <span>
                <span className="slds-text-heading_large">$30</span>
                <span className="slds-text-title">USD/user/month *</span>
              </span>
            </VisualPicker>
            <VisualPicker size="medium" label={PackageTwo}>
              <span>
                <span className="slds-text-heading_large">$150</span>
                <span className="slds-text-title">USD/user/month *</span>
              </span>
            </VisualPicker>
            <VisualPicker size="medium" disabled label={PackageThree}>
              <span>
                <span className="slds-text-heading_large">$300</span>
                <span className="slds-text-title">USD/user/month *</span>
              </span>
            </VisualPicker>
          </VisualPickerContainer>
        </FormElementControl>
      </Fieldset>
    )
  }
];
