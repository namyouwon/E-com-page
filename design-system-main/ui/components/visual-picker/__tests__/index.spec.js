/* eslint-env jest */
import React from 'react';
import { Fieldset, Legend } from '../../radio-group/base/example';
import { FormElementControl } from '../../form-element';
import {
  VisualPicker,
  VisualPickerContainer
} from '../coverable-content/example';
import {
  VisualPickerMediaObject,
  PackageOne,
  PackageTwo,
  PackageThree
} from '../non-coverable-content/example';
import createHelpers from '../../../../jest.helpers';

const { matchesMarkup } = createHelpers(__dirname);

it('renders a vertical visual picker', () =>
  matchesMarkup(
    <Fieldset>
      <Legend>Select an option</Legend>
      <FormElementControl>
        <VisualPickerContainer hasIcon={false}>
          <VisualPicker className="slds-visual-picker_vertical">
            <span>
              <span className="slds-text-heading_medium slds-m-bottom_x-small">
                Item Text
              </span>
              <span className="slds-text-title">
                Some optional item description to help the user better
                understand what this option is about.
              </span>
            </span>
          </VisualPicker>
          <VisualPicker className="slds-visual-picker_vertical">
            <span>
              <span className="slds-text-heading_medium slds-m-bottom_x-small">
                Item Text
              </span>
              <span className="slds-text-title">
                Some optional item description to help the user better
                understand what this option is about.
              </span>
            </span>
          </VisualPicker>
          <VisualPicker className="slds-visual-picker_vertical">
            <span>
              <span className="slds-text-heading_medium slds-m-bottom_x-small">
                Item Text
              </span>
              <span className="slds-text-title">
                Some optional item description to help the user better
                understand what this option is about.
              </span>
            </span>
          </VisualPicker>
        </VisualPickerContainer>
      </FormElementControl>
    </Fieldset>
  ));

it('renders a disabled vertical visual picker', () =>
  matchesMarkup(
    <Fieldset>
      <Legend>Select an option</Legend>
      <FormElementControl>
        <VisualPickerContainer hasIcon={false}>
          <VisualPicker className="slds-visual-picker_vertical">
            <span>
              <span className="slds-text-heading_medium slds-m-bottom_x-small">
                Item Text
              </span>
              <span className="slds-text-title">
                Some optional item description to help the user better
                understand what this option is about.
              </span>
            </span>
          </VisualPicker>
          <VisualPicker className="slds-visual-picker_vertical">
            <span>
              <span className="slds-text-heading_medium slds-m-bottom_x-small">
                Item Text
              </span>
              <span className="slds-text-title">
                Some optional item description to help the user better
                understand what this option is about.
              </span>
            </span>
          </VisualPicker>
          <VisualPicker className="slds-visual-picker_vertical" disabled>
            <span>
              <span className="slds-text-heading_medium slds-m-bottom_x-small">
                Item Text
              </span>
              <span className="slds-text-title">
                Some optional item description to help the user better
                understand what this option is about.
              </span>
            </span>
          </VisualPicker>
        </VisualPickerContainer>
      </FormElementControl>
    </Fieldset>
  ));

it('renders a coverable content visual picker', () =>
  matchesMarkup(
    <Fieldset>
      <Legend>Select an app</Legend>
      <FormElementControl>
        <VisualPickerContainer>
          <VisualPicker
            symbol="connected_apps"
            label="Connected App"
            size="medium"
          />
          <VisualPicker symbol="custom_apps" label="Custom App" size="medium" />
        </VisualPickerContainer>
      </FormElementControl>
    </Fieldset>
  ));

it('renders a disabled coverable content visual picker', () =>
  matchesMarkup(
    <Fieldset>
      <Legend>Select an app</Legend>
      <FormElementControl>
        <VisualPickerContainer type="checkbox">
          <VisualPicker
            symbol="connected_apps"
            label="Connected App"
            size="medium"
          />
          <VisualPicker
            disabled
            symbol="custom_apps"
            label="Custom App"
            size="medium"
          />
        </VisualPickerContainer>
      </FormElementControl>
    </Fieldset>
  ));

it('renders a coverable content visual picker with checkbox group', () =>
  matchesMarkup(
    <Fieldset>
      <Legend>Add the following object(s)</Legend>
      <FormElementControl>
        <VisualPickerContainer type="checkbox" sprite="standard">
          <VisualPicker symbol="account" label="Account" size="medium" />
          <VisualPicker symbol="lead" label="Lead" size="medium" />
          <VisualPicker symbol="orders" label="Orders" size="medium" />
        </VisualPickerContainer>
      </FormElementControl>
    </Fieldset>
  ));

it('renders a non-coverable content visual picker', () =>
  matchesMarkup(
    <Fieldset>
      <Legend>Select a plan</Legend>
      <FormElementControl>
        <VisualPickerContainer hasIcon={false}>
          <VisualPicker size="medium" label={PackageOne}>
            <span>
              <span className="slds-text-heading_large">$30</span>
              <span className="slds-text-title">USD/user/month *</span>
            </span>
          </VisualPicker>
          <VisualPicker size="medium" label={PackageTwo}>
            <span>
              <span className="slds-text-heading_large">$150</span>
              <span className="slds-text-title">USD/user/month *</span>
            </span>
          </VisualPicker>
          <VisualPicker size="medium" label={PackageThree}>
            <span>
              <span className="slds-text-heading_large">$300</span>
              <span className="slds-text-title">USD/user/month *</span>
            </span>
          </VisualPicker>
        </VisualPickerContainer>
      </FormElementControl>
    </Fieldset>
  ));

it('renders a disabled non-coverable content visual picker', () =>
  matchesMarkup(
    <Fieldset>
      <Legend>Select a plan</Legend>
      <FormElementControl>
        <VisualPickerContainer hasIcon={false}>
          <VisualPicker size="medium" label={PackageOne}>
            <span>
              <span className="slds-text-heading_large">$30</span>
              <span className="slds-text-title">USD/user/month *</span>
            </span>
          </VisualPicker>
          <VisualPicker size="medium" label={PackageTwo}>
            <span>
              <span className="slds-text-heading_large">$150</span>
              <span className="slds-text-title">USD/user/month *</span>
            </span>
          </VisualPicker>
          <VisualPicker size="medium" disabled label={PackageThree}>
            <span>
              <span className="slds-text-heading_large">$300</span>
              <span className="slds-text-title">USD/user/month *</span>
            </span>
          </VisualPicker>
        </VisualPickerContainer>
      </FormElementControl>
    </Fieldset>
  ));

it('renders a link visual picker', () =>
  matchesMarkup(
    <div className="demo-only" style={{ width: '24rem' }}>
      <VisualPickerMediaObject symbol="user">
        <h2
          className="slds-truncate slds-text-heading_small"
          title="Share the knowledge"
        >
          Share the knowledge
        </h2>
        <p className="slds-m-top_small">
          Harness your team's collective know-how with our powerful knowledge
          base
        </p>
      </VisualPickerMediaObject>
    </div>
  ));
