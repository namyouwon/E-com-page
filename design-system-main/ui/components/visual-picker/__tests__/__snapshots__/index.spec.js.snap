// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a coverable content visual picker 1`] = `
<fieldset
  className="slds-form-element"
>
  <legend
    className="slds-form-element__legend slds-form-element__label"
  >
    Select an app
  </legend>
  <div
    className="slds-form-element__control"
  >
    <div
      className="slds-visual-picker slds-visual-picker_medium"
    >
      <input
        id="visual-picker-7"
        name="example-unique-name-3"
        type="radio"
        value="visual-picker-7"
      />
      <label
        htmlFor="visual-picker-7"
      >
        <span
          className="slds-visual-picker__figure slds-visual-picker__icon slds-align_absolute-center"
        >
          <span
            className="slds-is-selected"
          >
            <span
              className="slds-icon_container"
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon_large slds-icon-action-check"
              >
                <use
                  xlinkHref="/assets/icons/action-sprite/svg/symbols.svg#check"
                />
              </svg>
            </span>
          </span>
          <span
            className="slds-is-not-selected"
          >
            <span
              className="slds-icon_container"
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon-utility-connected_apps slds-icon_large slds-icon-text-default"
              >
                <use
                  xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#connected_apps"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          className="slds-visual-picker__body"
        >
          Connected App
        </span>
      </label>
    </div>
    <div
      className="slds-visual-picker slds-visual-picker_medium"
    >
      <input
        id="visual-picker-8"
        name="example-unique-name-3"
        type="radio"
        value="visual-picker-8"
      />
      <label
        htmlFor="visual-picker-8"
      >
        <span
          className="slds-visual-picker__figure slds-visual-picker__icon slds-align_absolute-center"
        >
          <span
            className="slds-is-selected"
          >
            <span
              className="slds-icon_container"
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon_large slds-icon-action-check"
              >
                <use
                  xlinkHref="/assets/icons/action-sprite/svg/symbols.svg#check"
                />
              </svg>
            </span>
          </span>
          <span
            className="slds-is-not-selected"
          >
            <span
              className="slds-icon_container"
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon-utility-custom_apps slds-icon_large slds-icon-text-default"
              >
                <use
                  xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#custom_apps"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          className="slds-visual-picker__body"
        >
          Custom App
        </span>
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`renders a coverable content visual picker with checkbox group 1`] = `
<fieldset
  className="slds-form-element"
>
  <legend
    className="slds-form-element__legend slds-form-element__label"
  >
    Add the following object(s)
  </legend>
  <div
    className="slds-form-element__control"
  >
    <div
      className="slds-visual-picker slds-visual-picker_medium"
    >
      <input
        id="visual-picker-11"
        name="example-unique-name-5"
        type="checkbox"
        value="visual-picker-11"
      />
      <label
        htmlFor="visual-picker-11"
      >
        <span
          className="slds-visual-picker__figure slds-visual-picker__icon slds-align_absolute-center"
        >
          <span
            className="slds-is-selected"
          >
            <span
              className="slds-icon_container"
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon_large slds-icon-action-check"
              >
                <use
                  xlinkHref="/assets/icons/action-sprite/svg/symbols.svg#check"
                />
              </svg>
            </span>
          </span>
          <span
            className="slds-is-not-selected"
          >
            <span
              className="slds-icon_container"
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon-standard-account slds-icon_large"
              >
                <use
                  xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#account"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          className="slds-visual-picker__body"
        >
          Account
        </span>
      </label>
    </div>
    <div
      className="slds-visual-picker slds-visual-picker_medium"
    >
      <input
        id="visual-picker-12"
        name="example-unique-name-5"
        type="checkbox"
        value="visual-picker-12"
      />
      <label
        htmlFor="visual-picker-12"
      >
        <span
          className="slds-visual-picker__figure slds-visual-picker__icon slds-align_absolute-center"
        >
          <span
            className="slds-is-selected"
          >
            <span
              className="slds-icon_container"
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon_large slds-icon-action-check"
              >
                <use
                  xlinkHref="/assets/icons/action-sprite/svg/symbols.svg#check"
                />
              </svg>
            </span>
          </span>
          <span
            className="slds-is-not-selected"
          >
            <span
              className="slds-icon_container"
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon-standard-lead slds-icon_large"
              >
                <use
                  xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#lead"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          className="slds-visual-picker__body"
        >
          Lead
        </span>
      </label>
    </div>
    <div
      className="slds-visual-picker slds-visual-picker_medium"
    >
      <input
        id="visual-picker-13"
        name="example-unique-name-5"
        type="checkbox"
        value="visual-picker-13"
      />
      <label
        htmlFor="visual-picker-13"
      >
        <span
          className="slds-visual-picker__figure slds-visual-picker__icon slds-align_absolute-center"
        >
          <span
            className="slds-is-selected"
          >
            <span
              className="slds-icon_container"
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon_large slds-icon-action-check"
              >
                <use
                  xlinkHref="/assets/icons/action-sprite/svg/symbols.svg#check"
                />
              </svg>
            </span>
          </span>
          <span
            className="slds-is-not-selected"
          >
            <span
              className="slds-icon_container"
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon-standard-orders slds-icon_large"
              >
                <use
                  xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#orders"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          className="slds-visual-picker__body"
        >
          Orders
        </span>
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`renders a disabled coverable content visual picker 1`] = `
<fieldset
  className="slds-form-element"
>
  <legend
    className="slds-form-element__legend slds-form-element__label"
  >
    Select an app
  </legend>
  <div
    className="slds-form-element__control"
  >
    <div
      className="slds-visual-picker slds-visual-picker_medium"
    >
      <input
        id="visual-picker-9"
        name="example-unique-name-4"
        type="checkbox"
        value="visual-picker-9"
      />
      <label
        htmlFor="visual-picker-9"
      >
        <span
          className="slds-visual-picker__figure slds-visual-picker__icon slds-align_absolute-center"
        >
          <span
            className="slds-is-selected"
          >
            <span
              className="slds-icon_container"
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon_large slds-icon-action-check"
              >
                <use
                  xlinkHref="/assets/icons/action-sprite/svg/symbols.svg#check"
                />
              </svg>
            </span>
          </span>
          <span
            className="slds-is-not-selected"
          >
            <span
              className="slds-icon_container"
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon-utility-connected_apps slds-icon_large slds-icon-text-default"
              >
                <use
                  xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#connected_apps"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          className="slds-visual-picker__body"
        >
          Connected App
        </span>
      </label>
    </div>
    <div
      className="slds-visual-picker slds-visual-picker_medium"
    >
      <input
        disabled={true}
        id="visual-picker-10"
        name="example-unique-name-4"
        type="checkbox"
        value="visual-picker-10"
      />
      <label
        htmlFor="visual-picker-10"
      >
        <span
          className="slds-visual-picker__figure slds-visual-picker__icon slds-align_absolute-center"
        >
          <span
            className="slds-is-selected"
          >
            <span
              className="slds-icon_container"
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon_large slds-icon-action-check"
              >
                <use
                  xlinkHref="/assets/icons/action-sprite/svg/symbols.svg#check"
                />
              </svg>
            </span>
          </span>
          <span
            className="slds-is-not-selected"
          >
            <span
              className="slds-icon_container"
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon-utility-custom_apps slds-icon_large slds-icon-text-default"
              >
                <use
                  xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#custom_apps"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          className="slds-visual-picker__body"
        >
          Custom App
        </span>
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`renders a disabled non-coverable content visual picker 1`] = `
<fieldset
  className="slds-form-element"
>
  <legend
    className="slds-form-element__legend slds-form-element__label"
  >
    Select a plan
  </legend>
  <div
    className="slds-form-element__control"
  >
    <div
      className="slds-visual-picker slds-visual-picker_medium"
    >
      <input
        id="visual-picker-17"
        name="example-unique-name-7"
        type="radio"
        value="visual-picker-17"
      />
      <label
        htmlFor="visual-picker-17"
      >
        <span
          className="slds-visual-picker__figure slds-visual-picker__text slds-align_absolute-center"
        >
          <span>
            <span
              className="slds-text-heading_large"
            >
              $30
            </span>
            <span
              className="slds-text-title"
            >
              USD/user/month *
            </span>
          </span>
        </span>
        <span
          className="slds-visual-picker__body"
        >
          <span
            className="slds-text-heading_small"
          >
            Lightning Professional
          </span>
          <span
            className="slds-text-title"
          >
            Complete service CRM for teams of any size
          </span>
        </span>
        <span
          className="slds-icon_container slds-visual-picker__text-check"
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon-text-check slds-icon_x-small"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
            />
          </svg>
        </span>
      </label>
    </div>
    <div
      className="slds-visual-picker slds-visual-picker_medium"
    >
      <input
        id="visual-picker-18"
        name="example-unique-name-7"
        type="radio"
        value="visual-picker-18"
      />
      <label
        htmlFor="visual-picker-18"
      >
        <span
          className="slds-visual-picker__figure slds-visual-picker__text slds-align_absolute-center"
        >
          <span>
            <span
              className="slds-text-heading_large"
            >
              $150
            </span>
            <span
              className="slds-text-title"
            >
              USD/user/month *
            </span>
          </span>
        </span>
        <span
          className="slds-visual-picker__body"
        >
          <span
            className="slds-text-heading_small"
          >
            Lightning Enterprise
          </span>
          <span
            className="slds-text-title"
          >
            Everything you need to take support to the next level
          </span>
        </span>
        <span
          className="slds-icon_container slds-visual-picker__text-check"
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon-text-check slds-icon_x-small"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
            />
          </svg>
        </span>
      </label>
    </div>
    <div
      className="slds-visual-picker slds-visual-picker_medium"
    >
      <input
        disabled={true}
        id="visual-picker-19"
        name="example-unique-name-7"
        type="radio"
        value="visual-picker-19"
      />
      <label
        htmlFor="visual-picker-19"
      >
        <span
          className="slds-visual-picker__figure slds-visual-picker__text slds-align_absolute-center"
        >
          <span>
            <span
              className="slds-text-heading_large"
            >
              $300
            </span>
            <span
              className="slds-text-title"
            >
              USD/user/month *
            </span>
          </span>
        </span>
        <span
          className="slds-visual-picker__body"
        >
          <span
            className="slds-text-heading_small"
          >
            Lightning Unlimited
          </span>
          <span
            className="slds-text-title"
          >
            Complete support with enterprise-grade customization
          </span>
        </span>
        <span
          className="slds-icon_container slds-visual-picker__text-check"
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon-text-check slds-icon_x-small"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
            />
          </svg>
        </span>
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`renders a disabled vertical visual picker 1`] = `
<fieldset
  className="slds-form-element"
>
  <legend
    className="slds-form-element__legend slds-form-element__label"
  >
    Select an option
  </legend>
  <div
    className="slds-form-element__control"
  >
    <div
      className="slds-visual-picker slds-visual-picker_vertical"
    >
      <input
        id="visual-picker-4"
        name="example-unique-name-2"
        type="radio"
        value="visual-picker-4"
      />
      <label
        htmlFor="visual-picker-4"
      >
        <span
          className="slds-visual-picker__figure slds-visual-picker__text slds-align_absolute-center"
        >
          <span>
            <span
              className="slds-text-heading_medium slds-m-bottom_x-small"
            >
              Item Text
            </span>
            <span
              className="slds-text-title"
            >
              Some optional item description to help the user better understand what this option is about.
            </span>
          </span>
        </span>
        <span
          className="slds-icon_container slds-visual-picker__text-check"
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon-text-check slds-icon_x-small"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
            />
          </svg>
        </span>
      </label>
    </div>
    <div
      className="slds-visual-picker slds-visual-picker_vertical"
    >
      <input
        id="visual-picker-5"
        name="example-unique-name-2"
        type="radio"
        value="visual-picker-5"
      />
      <label
        htmlFor="visual-picker-5"
      >
        <span
          className="slds-visual-picker__figure slds-visual-picker__text slds-align_absolute-center"
        >
          <span>
            <span
              className="slds-text-heading_medium slds-m-bottom_x-small"
            >
              Item Text
            </span>
            <span
              className="slds-text-title"
            >
              Some optional item description to help the user better understand what this option is about.
            </span>
          </span>
        </span>
        <span
          className="slds-icon_container slds-visual-picker__text-check"
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon-text-check slds-icon_x-small"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
            />
          </svg>
        </span>
      </label>
    </div>
    <div
      className="slds-visual-picker slds-visual-picker_vertical"
    >
      <input
        disabled={true}
        id="visual-picker-6"
        name="example-unique-name-2"
        type="radio"
        value="visual-picker-6"
      />
      <label
        htmlFor="visual-picker-6"
      >
        <span
          className="slds-visual-picker__figure slds-visual-picker__text slds-align_absolute-center"
        >
          <span>
            <span
              className="slds-text-heading_medium slds-m-bottom_x-small"
            >
              Item Text
            </span>
            <span
              className="slds-text-title"
            >
              Some optional item description to help the user better understand what this option is about.
            </span>
          </span>
        </span>
        <span
          className="slds-icon_container slds-visual-picker__text-check"
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon-text-check slds-icon_x-small"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
            />
          </svg>
        </span>
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`renders a link visual picker 1`] = `
<div
  className="demo-only"
  style={
    Object {
      "width": "24rem",
    }
  }
>
  <a
    className="slds-box slds-box_link slds-box_x-small slds-media"
    href="#"
    onClick={[Function]}
  >
    <div
      className="slds-media__figure slds-media__figure_fixed-width slds-align_absolute-center slds-m-left_xx-small"
    >
      <span
        className="slds-icon_container slds-icon-utility-knowledge_base"
        title="Description of icon when needed"
      >
        <svg
          aria-hidden={true}
          className="slds-icon slds-icon-text-default"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#knowledge_base"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Description of icon when needed
        </span>
      </span>
    </div>
    <div
      className="slds-media__body slds-border_left slds-p-around_small"
    >
      <h2
        className="slds-truncate slds-text-heading_small"
        title="Share the knowledge"
      >
        Share the knowledge
      </h2>
      <p
        className="slds-m-top_small"
      >
        Harness your team's collective know-how with our powerful knowledge base
      </p>
    </div>
  </a>
</div>
`;

exports[`renders a non-coverable content visual picker 1`] = `
<fieldset
  className="slds-form-element"
>
  <legend
    className="slds-form-element__legend slds-form-element__label"
  >
    Select a plan
  </legend>
  <div
    className="slds-form-element__control"
  >
    <div
      className="slds-visual-picker slds-visual-picker_medium"
    >
      <input
        id="visual-picker-14"
        name="example-unique-name-6"
        type="radio"
        value="visual-picker-14"
      />
      <label
        htmlFor="visual-picker-14"
      >
        <span
          className="slds-visual-picker__figure slds-visual-picker__text slds-align_absolute-center"
        >
          <span>
            <span
              className="slds-text-heading_large"
            >
              $30
            </span>
            <span
              className="slds-text-title"
            >
              USD/user/month *
            </span>
          </span>
        </span>
        <span
          className="slds-visual-picker__body"
        >
          <span
            className="slds-text-heading_small"
          >
            Lightning Professional
          </span>
          <span
            className="slds-text-title"
          >
            Complete service CRM for teams of any size
          </span>
        </span>
        <span
          className="slds-icon_container slds-visual-picker__text-check"
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon-text-check slds-icon_x-small"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
            />
          </svg>
        </span>
      </label>
    </div>
    <div
      className="slds-visual-picker slds-visual-picker_medium"
    >
      <input
        id="visual-picker-15"
        name="example-unique-name-6"
        type="radio"
        value="visual-picker-15"
      />
      <label
        htmlFor="visual-picker-15"
      >
        <span
          className="slds-visual-picker__figure slds-visual-picker__text slds-align_absolute-center"
        >
          <span>
            <span
              className="slds-text-heading_large"
            >
              $150
            </span>
            <span
              className="slds-text-title"
            >
              USD/user/month *
            </span>
          </span>
        </span>
        <span
          className="slds-visual-picker__body"
        >
          <span
            className="slds-text-heading_small"
          >
            Lightning Enterprise
          </span>
          <span
            className="slds-text-title"
          >
            Everything you need to take support to the next level
          </span>
        </span>
        <span
          className="slds-icon_container slds-visual-picker__text-check"
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon-text-check slds-icon_x-small"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
            />
          </svg>
        </span>
      </label>
    </div>
    <div
      className="slds-visual-picker slds-visual-picker_medium"
    >
      <input
        id="visual-picker-16"
        name="example-unique-name-6"
        type="radio"
        value="visual-picker-16"
      />
      <label
        htmlFor="visual-picker-16"
      >
        <span
          className="slds-visual-picker__figure slds-visual-picker__text slds-align_absolute-center"
        >
          <span>
            <span
              className="slds-text-heading_large"
            >
              $300
            </span>
            <span
              className="slds-text-title"
            >
              USD/user/month *
            </span>
          </span>
        </span>
        <span
          className="slds-visual-picker__body"
        >
          <span
            className="slds-text-heading_small"
          >
            Lightning Unlimited
          </span>
          <span
            className="slds-text-title"
          >
            Complete support with enterprise-grade customization
          </span>
        </span>
        <span
          className="slds-icon_container slds-visual-picker__text-check"
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon-text-check slds-icon_x-small"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
            />
          </svg>
        </span>
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`renders a vertical visual picker 1`] = `
<fieldset
  className="slds-form-element"
>
  <legend
    className="slds-form-element__legend slds-form-element__label"
  >
    Select an option
  </legend>
  <div
    className="slds-form-element__control"
  >
    <div
      className="slds-visual-picker slds-visual-picker_vertical"
    >
      <input
        id="visual-picker-1"
        name="example-unique-name-1"
        type="radio"
        value="visual-picker-1"
      />
      <label
        htmlFor="visual-picker-1"
      >
        <span
          className="slds-visual-picker__figure slds-visual-picker__text slds-align_absolute-center"
        >
          <span>
            <span
              className="slds-text-heading_medium slds-m-bottom_x-small"
            >
              Item Text
            </span>
            <span
              className="slds-text-title"
            >
              Some optional item description to help the user better understand what this option is about.
            </span>
          </span>
        </span>
        <span
          className="slds-icon_container slds-visual-picker__text-check"
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon-text-check slds-icon_x-small"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
            />
          </svg>
        </span>
      </label>
    </div>
    <div
      className="slds-visual-picker slds-visual-picker_vertical"
    >
      <input
        id="visual-picker-2"
        name="example-unique-name-1"
        type="radio"
        value="visual-picker-2"
      />
      <label
        htmlFor="visual-picker-2"
      >
        <span
          className="slds-visual-picker__figure slds-visual-picker__text slds-align_absolute-center"
        >
          <span>
            <span
              className="slds-text-heading_medium slds-m-bottom_x-small"
            >
              Item Text
            </span>
            <span
              className="slds-text-title"
            >
              Some optional item description to help the user better understand what this option is about.
            </span>
          </span>
        </span>
        <span
          className="slds-icon_container slds-visual-picker__text-check"
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon-text-check slds-icon_x-small"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
            />
          </svg>
        </span>
      </label>
    </div>
    <div
      className="slds-visual-picker slds-visual-picker_vertical"
    >
      <input
        id="visual-picker-3"
        name="example-unique-name-1"
        type="radio"
        value="visual-picker-3"
      />
      <label
        htmlFor="visual-picker-3"
      >
        <span
          className="slds-visual-picker__figure slds-visual-picker__text slds-align_absolute-center"
        >
          <span>
            <span
              className="slds-text-heading_medium slds-m-bottom_x-small"
            >
              Item Text
            </span>
            <span
              className="slds-text-title"
            >
              Some optional item description to help the user better understand what this option is about.
            </span>
          </span>
        </span>
        <span
          className="slds-icon_container slds-visual-picker__text-check"
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon-text-check slds-icon_x-small"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
            />
          </svg>
        </span>
      </label>
    </div>
  </div>
</fieldset>
`;
